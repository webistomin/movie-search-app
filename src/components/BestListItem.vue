<template>
  <li class="best__item">
    <a href="" class="best__link">
      <picture class="best__pic best__pic--gold">
        <img src="/static/img/content/best@1x.jpg" alt="#" width="41" height="41" class="best__img">
      </picture>
      <div class="best__block best__block--padding">
        <span class="best__name">Gravity</span>
        <div class="best__rating"></div>
      </div>
    </a>
  </li>
</template>

<script>
  export default {
    name: 'BestItem',
  };
</script>

<style>

</style>
