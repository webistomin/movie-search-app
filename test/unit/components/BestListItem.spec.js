import { shallowMount, createLocalVue } from '@vue/test-utils';
import BestListItem from '../../../src/components/BestListItem';

describe('BestListItem', () => {
  let vm;
  let localVue;

  beforeEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
    localVue = createLocalVue();
  });

  it('Matches snapshot', () => {
    vm = shallowMount(BestListItem, {
      localVue,
      propsData: {
        movieTitle: 'Prison Break',
        movieId: 0,
        movieCount: 5,
        moviePoster: 'https://pmctvline2.files.wordpress.com/2016/01/prison-break-returns-fox.jpg?w=620&h=420&crop=1',
        movieRate: 10,
      },
      stubs: ['router-link', 'router-view'],
    }).vm;
    expect(vm.$el).toMatchSnapshot();
  });

  describe('get right border color', () => {
    it('get silver color if movieCount <= 3', () => {
      vm = shallowMount(BestListItem, {
        localVue,
        propsData: {
          movieTitle: 'Prison Break',
          movieId: 0,
          movieCount: 3,
          moviePoster: 'https://pmctvline2.files.wordpress.com/2016/01/prison-break-returns-fox.jpg?w=620&h=420&crop=1',
          movieRate: 10,
        },
        stubs: ['router-link', 'router-view'],
      }).vm;
      expect(vm.getBorderColor).toEqual('best__pic--silver');
    });

    it('get gold color if movieCount > 3', () => {
      vm = shallowMount(BestListItem, {
        localVue,
        propsData: {
          movieTitle: 'Prison Break',
          movieId: 0,
          movieCount: 1,
          moviePoster: 'https://pmctvline2.files.wordpress.com/2016/01/prison-break-returns-fox.jpg?w=620&h=420&crop=1',
          movieRate: 10,
        },
        stubs: ['router-link', 'router-view'],
      }).vm;
      expect(vm.getBorderColor).toEqual('best__pic--gold');
    });
  });

  describe('get right poster src', () => {
    it('get movie poster if moviePoster props exist', () => {
      vm = shallowMount(BestListItem, {
        localVue,
        propsData: {
          movieTitle: 'Prison Break',
          movieId: 0,
          movieCount: 3,
          moviePoster: '/prison-break-returns-fox.jpg?w=620&h=420&crop=1',
          movieRate: 10,
        },
        stubs: ['router-link', 'router-view'],
      }).vm;
      expect(vm.getPosterSrc).toEqual(`https://image.tmdb.org/t/p/w92/${vm.moviePoster}`);
    });

    it('get static movie poster if moviePoster props doesnt exist', () => {
      vm = shallowMount(BestListItem, {
        localVue,
        propsData: {
          movieTitle: 'Prison Break',
          movieId: 0,
          movieCount: 3,
          moviePoster: '',
          movieRate: 10,
        },
        stubs: ['router-link', 'router-view'],
      }).vm;
      expect(vm.getPosterSrc).toEqual('/static/img/content/image-not-found.svg');
    });
  });
});
