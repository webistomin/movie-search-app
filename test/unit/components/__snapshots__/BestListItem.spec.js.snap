// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestListItem Matches snapshot 1`] = `
<li
  class="best__item"
>
  <router-link-stub
    class="best__link"
    to="/movie/0"
  >
    <picture
      class="best__pic best__pic--silver"
    >
      <img
        alt="Prison Break"
        class="best__img"
        height="41"
        src="https://image.tmdb.org/t/p/w92/https://pmctvline2.files.wordpress.com/2016/01/prison-break-returns-fox.jpg?w=620&h=420&crop=1"
        width="41"
      />
    </picture>
     
    <div
      class="best__block best__block--padding"
    >
      <span
        class="best__name"
      >
        Prison Break
      </span>
       
      <span
        class="best__rate"
      >
        <svg
          class="best__icon"
          height="12"
          width="12"
        >
          <use
            xlink:href="#icon-like"
          />
        </svg>
        
        10
    
      </span>
    </div>
  </router-link-stub>
</li>
`;
